src/posix/platform/CMakeLists.txt

```diff
@@ -146,6 +146,7 @@ add_library(openthread-posix
     rcp_caps_diag.cpp
     resolver.cpp
     settings.cpp
+    settings_file.cpp
```

Comment:
Do we need to also add this to BUILD.gn file? (Though we don't use it very often)

Reply:
The source files under the path `src/posix` do not have the BUILD.gn file. There is no BUILD.gn file for us to modify.